presto-main/src/main/java/com/facebook/presto/sql/gen/RowExpressionCompiler.java

```diff
@@ -210,9 +212,7 @@ public BytecodeNode visitLambda(LambdaDefinitionExpression lambda, Context conte
         {
             checkState(preGeneratedExpressions.getCompiledLambdaMetadata().containsKey(lambda), "lambda expressions map does not contain this lambda definition");
             if (MethodHandle.class.equals(context.getLambdaInterface().get())) {
-
-                return context.getScope().getThis().getField(preGeneratedExpressions.getCompiledLambdaMetadata().get(lambda).getInstanceField())
-                        .invoke("bindTo", MethodHandle.class, context.getScope().getVariable("session").cast(Object.class));
+                throw new PrestoException(FUNCTION_IMPLEMENTATION_ERROR, "Lambda cannot be generated as MethodHandle");
```

Comment:
If this is impossible, throw a different exception (like VerifyException)

In the exception (or comments), make it explicit that this is should not happen due to prior sanity checks.